src/reporters/BroadcastReporter.ts

```diff
@@ -11,25 +13,21 @@ export interface NamedReporter {
   reporter: Reporter;
 }
 
-export const ALL_EVENT_METHOD_NAMES = ['onSourceFileRead', 'onAllSourceFilesRead', 'onAllMutantsMatchedWithTests', 'onMutantTested', 'onAllMutantsTested', 'onConfigRead'];
+export const ALL_EVENT_METHOD_NAMES = ['onSourceFileRead', 'onAllSourceFilesRead', 'onAllMutantsMatchedWithTests', 'onMutantTested', 'onAllMutantsTested'];
 
-export default class BroadcastReporter implements Reporter {
+export default class BroadcastReporter implements StrictReporter {
 
   constructor(private reporters: NamedReporter[]) {
-    ALL_EVENT_METHOD_NAMES.concat('wrapUp').forEach(method => {
-      (<any>this)[method] = (arg: any) => {
-        return this.broadcast(method, arg);
-      };
-    });
+    // Empty constructor
   }
 
-  private broadcast(methodName: string, eventArgs: any) {
+  private broadcast(methodName: string, eventArgs?: any) {
     let allPromises: Promise<any>[] = [];
     this.reporters.forEach(namedReporter => {
       let reporter = <any>namedReporter.reporter;
       if (reporter[methodName] && typeof reporter[methodName] === 'function') {
         try {
-          let maybePromise: void | Promise<any> = reporter[methodName](eventArgs);
+          let maybePromise: void | Promise<any> = eventArgs ? reporter[methodName](eventArgs) : reporter[methodName]();
```

Comment:
I will look into this sometime soon, I think it's done because there is no certainty whether `eventArgs` is an actual value or `undefined` / `null`.
If there is a possibility that `eventArgs` is `undefined` / `null` the typescript compiler will complain with the newly added flags.